app/controllers/hosts_controller.rb

```diff
@@ -546,7 +546,8 @@ def process_taxonomy
   end
 
   def template_used
-    host      = Host.new(clean_interfaces_attributes.except(:host_parameters_attributes))
+    host = params[:id] ? Host.find(params[:id]) : Host.new
```

Comment:
Using the real host has an unfortunate side effect that calling `attributes=` below will immediate change relations such as Puppet classes.  Try taking an existing host, load #edit, add a Puppet class to it and click Resolve, it'll get added in the DB.